pandas/core/arrays/period.py

```diff
@@ -228,6 +303,183 @@ def start_time(self):
     def end_time(self):
         return self.to_timestamp(how='end')
 
+    def __repr__(self):
+        return '<{}>\n{}\nLength: {}, dtype: {}'.format(
+            self.__class__.__name__,
+            [str(s) for s in self],
+            len(self),
+            self.dtype
+        )
+
+    def __len__(self):
+        return len(self._data)
+
+    def __setitem__(self, key, value):
+        if is_list_like(value):
```

Comment:
can you add types on key / value

Reply:
Done. The syntax is for wrapping long lines is kind of ugly :/

```
    def __setitem__(
            self,
            key,   # type: Union[int, Sequence[int], Sequence[bool]]
            value  # type: Union[NaTType, Period, Sequence[Period]]
        ):
        # type: (...) -> None
```


this type is